react/src/util/opReturn.ts

```diff
@@ -1,20 +1,69 @@
-export const OP_RETURN_PREFIX = 'paybutton$';
+import { lib, enc } from 'crypto-js';
 
-export function stringToHex(str: string): string {
+//opReturn format is:
+//`04` - Pushdata of the Protocol Identifier, `50415900`
+//`50415900` - PayButton protocol identifier, ascii "PAY"
+//`00` - Version 0, pushed as `OP_0`
+//`09` - pushdata for the data payload, signifying this tx has 9 bytes of data
+//`0102030405060708aa` - data payload
+//`08` - pushdata for the optional nonce payload, signifying this tx has 8 bytes of nonce data
+//`0102030405060708` - The 8-byte nonce
+
+// All the below variables are already encoded to HEX
+export const OP_RETURN_PREFIX_PUSH_DATA = '04'; // \x04
+export const OP_RETURN_PREFIX = '50415900'; // PAY\x00
+export const VERSION = '00'; // \x00
+
+// 220 total bytes
+// - 5 from the 4-byte procol prefix with push data: '\x04PAY\x00'
+// - 1 for the version byte: '\x00'
+// - 9 from the 8-byte nonce with push data
+// = 205 available bytes
+export const BYTES_LIMIT = 220 - 5 - 1 - 9; // 205
+
+function stringToHex(str: string): string {
   return str
     .split('')
     .map(c => c.charCodeAt(0).toString(16).padStart(2, '0'))
     .join('');
 }
 
-export function parseOpReturn(
+function generatePushDataPrefixed8ByteNonce(): string {
+  // Generate 8 random bytes
+  const wordArray = lib.WordArray.random(8);
+
+  // Convert the word array to a hex string
+  const hexString = enc.Hex.stringify(wordArray);
+
+  // The result is 18 char long:
+  // ---
+  // a hex character encodes 4 bits of information (2⁴ = 16);
+  // ... therefore; 8 bytes = 72 bits => 72/4 = 16 hex chars
+  // + 1 byte of push data at the beggining = 2 hex chars
+  // = 18 chars
+  return `08${hexString}`;
```

Comment:
It is perhaps nice to avoid using a library like `utxolib` or the `Buffer` object, which Cashtab uses to create OP_RETURN outputs.

However, if you are going to avoid those libs, would still recommend a more flexible way of creating OP_RETURN outputs. 

A function that only makes a substring for a certain type of spec is not very useful if the spec ever changes.  Ideally you have a function like the `script.push` [here](https://github.com/Bitcoin-ABC/bitcoin-abc/blob/master/cashtab/src/opreturn/index.js#L141), where the correct pushdata is added automatically based on the data you are pushing.

Reply:
Yeah, `Buffer ` doesn't work well when bundling.

I created a function to prepend a pushData to a hex string, more clean indeed.